crates/extensions/c8y_firmware_manager/src/actor.rs

```diff
@@ -248,7 +248,8 @@ impl FirmwareManagerActor {
             let auth = if self
                 .config
                 .c8y_end_point
-                .url_is_in_my_tenant_domain(firmware_url)
+                .maybe_tenant_url(firmware_url)
+                .is_some()
             {
                 if let Ok(token) = self.message_box.jwt_retriever.await_response(()).await? {
                     Some(Auth::new_bearer(&token))
```

Comment:
This is out of scope for this PR, but we need to consider using `c8y_auth_proxy` here instead of managing jwt tokens. This will be required when moving firmware support to a tedge plugin and no more a c8y plugin.